### PR TITLE
add ledstrip

### PR DESCRIPTION
**Pull request by Frincu13**

### CODE REVIEW

**Review by Belereg — APPROVED** [review]

File: CoffeeMachineController.cpp [hunk 1]
```diff
@@ -169,6 +176,69 @@ class CoffeeMachineController
     response.send(Http::Code::Ok, res.dump(4));
   }
 
+  void LedStripState(const Rest::Request &request, Http::ResponseWriter response)
+  {
+    json res;
+    // We can see how dirty the coffee machine is before cleaning it
```
Comment (andreibudoi) [inline CoffeeMachineController.cpp]: comentariu irelevant

File: CoffeeMachineController.cpp [hunk 2]
```diff
@@ -71,6 +74,10 @@ class CoffeeMachineController
     // Clean coffee machine
     Routes::Get(router, "/getCleanLevel", Routes::bind(&CoffeeMachineController::cleanLevel, this));
     Routes::Post(router, "/cleanCoffeeMachine", Routes::bind(&CoffeeMachineController::clean, this));
+
+    // Led strip controller
+    Routes::Get(router, "/getLedStrip", Routes::bind(&CoffeeMachineController::LedStripState, this));
```
Comment (andreibudoi) [inline CoffeeMachineController.cpp]: Functii cu litera mare, nasol

File: CoffeeMachineController.cpp [hunk 3]
```diff
@@ -169,6 +176,69 @@ class CoffeeMachineController
...
+  {
+    json res;
+    // We can see how dirty the coffee machine is before cleaning it
+    string color;
+    bool state = coffeeMachine.getLedStripState();
+    if (state == false)
+    {
+      res["status"] = "LedStrip is off";
+    }
+    else
```
Comment (andreibudoi) [inline CoffeeMachineController.cpp:379]: if else state == true

File: CoffeeMachineController.cpp [hunk 4]
```diff
@@ -169,6 +176,69 @@ class CoffeeMachineController
...
+          }
+          else 
+          {
+            coffeeMachine.setLedStripColor(color);
+            res["status"]="LedStrip is on with color "+color;
+          }
+    }
+    else
+    {
+      res["status"] = "Input invalid";
```
Comment (Belereg) [inline CoffeeMachineController.cpp]: ar trebui sa returnezi un status code 400 bad request in cazul unui bad input

File: CoffeeMachineController.cpp [hunk 5]
```diff
@@ -169,6 +176,69 @@ class CoffeeMachineController
...
+    json req = json::parse(request.body());
+    cout << req.dump(4); //4 spaces as tab in json
+
+    string color=req["color"];
+    string state = req["state"];
+
+    json res;
+    
+    
+    if (regex_match (state, regex("[0-1]") ) && regex_match (color, regex("#[a-zA-Z0-9]{6}") ))
```
Comment (Belereg) [inline CoffeeMachineController.cpp]: valorile hex contin litere de la 'a' la 'f', nu pana la 'z'

**Review by Belereg — CHANGES_REQUESTED** [review]

**Review by andreibudoi — CHANGES_REQUESTED** [review]

File: CoffeeMachineController.cpp (hunk 3 above)
Comment (Frincu13) [reply CoffeeMachineController.cpp:379]: not needed

File: CoffeeMachineController.cpp (hunk 2 above)
Comment (Frincu13) [reply CoffeeMachineController.cpp]: done

File: CoffeeMachineController.cpp (hunk 1 above)
Comment (Frincu13) [reply CoffeeMachineController.cpp]: yes sir

File: CoffeeMachineController.cpp (hunk 5 above)
Comment (Frincu13) [reply CoffeeMachineController.cpp]: thx,that's true

File: CoffeeMachineController.cpp (hunk 4 above)
Comment (Frincu13) [reply CoffeeMachineController.cpp]: added.thx

**Review by Belereg — APPROVED** [review]

**Review by andreibudoi — APPROVED** [review]

**Pull request merged**